sys/psa_crypto/psa_key_slot_mgmt/psa_crypto_slot_management.c

```diff
@@ -0,0 +1,529 @@
+    DEBUG("Asymmetric Slot Array Size: %d\n", sizeof(key_pair_slots));
+    DEBUG("Asymmetric Slot Empty List Size: %d\n", clist_count(&key_pair_list_empty));
+#endif /* MODULE_PSA_ASYMMETRIC */
+
+    memset(single_key_slots, 0, sizeof(single_key_slots));
+#if PSA_SINGLE_KEY_COUNT
+    for (size_t i = 0; i < PSA_SINGLE_KEY_COUNT; i++) {
+        clist_rpush(&single_key_list_empty, &single_key_slots[i].node);
+    }
+#endif
+    DEBUG("Single Key Slot Count: %d, Size: %d\n", PSA_SINGLE_KEY_COUNT, sizeof(psa_key_slot_t));
+    DEBUG("Single Key Slot Array Size: %d\n", sizeof(single_key_slots));
+    DEBUG("Single Key Slot Empty List Size: %d\n", clist_count(&single_key_list_empty));
+}
+
+/**
+ * @brief   Wipe key slot with correct key slot size
+ *
+ * @param   slot    Key sloit to be wiped
+ */
+static void psa_wipe_real_slot_type(psa_key_slot_t *slot)
+{
+    psa_key_attributes_t attr = slot->attr;
+
+    if (!psa_key_lifetime_is_external(attr.lifetime)) {
+        if (!PSA_KEY_TYPE_IS_KEY_PAIR(attr.type)) {
+            memset(slot, 0, sizeof(psa_key_slot_t));
+        }
+#if IS_USED(MODULE_PSA_ASYMMETRIC)
+        else {
+
+            memset((psa_key_pair_slot_t *)slot, 0, sizeof(psa_key_pair_slot_t));
+        }
+#endif
+    }
+#if IS_USED(MODULE_PSA_SECURE_ELEMENT)
+    else {
+        memset((psa_prot_key_slot_t *)slot, 0, sizeof(psa_prot_key_slot_t));
+    }
+#endif
+}
+
+psa_status_t psa_wipe_key_slot(psa_key_slot_t *slot)
+{
+    /* Get list the slot is stored in */
+    clist_node_t *empty_list = psa_get_empty_key_slot_list(&slot->attr);
+
+    if (empty_list == NULL) {
+        return PSA_ERROR_DOES_NOT_EXIST;
+    }
+
+    /* Get node to remove from key slot list */
+    clist_node_t *n = clist_remove(&key_slot_list, &slot->node);
+
+    if (n == NULL) {
+        return PSA_ERROR_DOES_NOT_EXIST;
+    }
+
+    psa_key_slot_t *tmp = container_of(n, psa_key_slot_t, node);
+
+    /* Wipe slot associated with node */
+    psa_wipe_real_slot_type(tmp);
+
+    /* Append node to empty list for later reuse */
+    clist_rpush(empty_list, n);
+    return PSA_SUCCESS;
+}
+
+void psa_wipe_all_key_slots(void)
+{
+    /* Move all list items to empty lists */
+    while (!clist_is_empty(&key_slot_list)) {
+        clist_node_t *to_remove = clist_rpop(&key_slot_list);
+        psa_key_slot_t *slot = container_of(to_remove, psa_key_slot_t, node);
+        clist_node_t *empty_list = psa_get_empty_key_slot_list(&slot->attr);
+
+        psa_wipe_real_slot_type(slot);
+        clist_rpush(empty_list, to_remove);
+    }
+}
+
+/**
+ * @brief   Find a key slot with the desired key ID in key slot list
+ *
+ *          This is a function used by the @ref clist_foreach function.
+ *
+ * @param   n   Pointer to clist node referencing the required key slot
+ * @param   arg ID of the required key
+ * @return  int
+ *          1 if the key is found
+ *          0 if the key is not found
+ */
+static int psa_get_node_with_id(clist_node_t *n, void *arg)
```

Comment:
I think this is a bit hard to understand: The arguments, function name and doc description. Could you rephrase that?